### PR TITLE
Ported more operators to ReductionOpInfo testing

### PR DESCRIPTION
**Pull request by heitorschueroff**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63035
* #62920
* #62900
* #62899
* #62737

### CODE REVIEW

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -5219,18 +5219,38 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+
+        if 'dim' in keys:
+            dim = kwargs.pop('dim')
+
+            # NumPy reductions don't accept dim=0 for scalar inputs
+            kwargs['axis'] = None if t.ndim == 0 and (dim == 0 or dim == -1) else dim
+
+        if 'keepdim' in keys:
+            if not supports_keepdims:
+                # NumPy does not support keepdims kwarg so we manually unsqueeze result
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:5235]: Cool

File: torch/testing/_internal/common_methods_invocations.py [hunk 2]
```diff
@@ -5219,18 +5219,38 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+                if kwargs.pop('keepdim'):
+                    res = f(t.cpu().numpy(), *args, **kwargs)
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:5237]: detach()

File: torch/testing/_internal/common_methods_invocations.py [hunk 3]
```diff
@@ -5219,18 +5219,38 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
+                    dims = kwargs.get('axis', list(range(t.ndim)))
+                    return np.expand_dims(res, dims)
+            else:
+                kwargs['keepdims'] = kwargs.pop('keepdim')
 
-        return f(t.cpu().numpy(), *args, axis=dim, keepdims=keepdim, **kwargs)
+        return f(t.cpu().numpy(), *args, **kwargs)
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:5243]: detach()

File: torch/testing/_internal/common_methods_invocations.py [hunk 4]
```diff
@@ -5219,18 +5219,38 @@ def gradcheck_wrapper_triangular_input(op, input, *args, upper=False, **kwargs):
...
 
     return wrapper
 
 
+def reference_logsumexp(t, dim=None, keepdim=False):
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:5248]: Cool reference; the version without SciPy may have numerical stability issues but I don't think our CI runs without SciPy. Another option would be to only define the reference if SciPy is available.

File: torch/testing/_internal/common_methods_invocations.py [hunk 5]
```diff
@@ -8575,6 +8600,30 @@ def wrapper(t: torch.Tensor, *args, dim=None, keepdim=False, **kwargs):
...
+        identity=-inf,
+        nan_policy='propagate',
+        promotes_int_to_float=True,
+        assert_autodiffed=True,
+        dtypes=floating_types_and(torch.bfloat16),
+        dtypesIfCUDA=floating_types_and(torch.bfloat16, torch.half),
+        sample_inputs_func=sample_inputs_logsumexp,
+        reference=reference_logsumexp,
+        skips=(
+            # dim parameter is not optional
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:8615]: there's an issue for this, right?

**Review by mruberry — COMMENTED** [review]
Looks really good; does this avoid porting aminmax because it returns two tensors and the tests don't handle that?

**Comment by heitorschueroff** [comment]
I merged this into the previous PR on the stack and undid the port for logsumexp because it requires some more thought with the identity value.

**Pull request closed without merging**